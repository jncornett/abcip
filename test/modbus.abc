d ( stack="eth:ip4:tcp:modbus" )
c ( tcp:b=502 )

a ( tcp:syn )
b ( tcp:syn, ack )
a ( tcp:ack )

a ( tcp:ack, modbus:func=3, pay="|03 00 04 00 01|" )

