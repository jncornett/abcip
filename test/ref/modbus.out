IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.502: Flags [S], cksum 0xb1eb (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 01f6 0000 0001 0000 0000 5002  ..............P.
	0x0030:  2000 b1eb 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.502 > 10.1.2.3.48620: Flags [S.], cksum 0xb1d9 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 01f6 bdec 0000 0001 0000 0002 5012  ..............P.
	0x0030:  2000 b1d9 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.502: Flags [.], cksum 0xb1da (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 01f6 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 b1da 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 53)
    10.1.2.3.48620 > 10.9.8.7.502: Flags [.], cksum 0xaac1 (correct), seq 1:14, ack 1, win 8192, length 13
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0035 0003 0000 4006 5cad 0a01 0203 0a09  .5....@.\.......
	0x0020:  0807 bdec 01f6 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 aac1 0000 0001 0000 0007 ff03 0300  ................
	0x0040:  0400 01                                  ...
